src/Features/CSharp/Portable/CodeFixes/MakeStatementAsynchronous/CSharpMakeStatementAsynchronousCodeFixProvider.cs

```diff
@@ -0,0 +1,119 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.MakeStatementAsynchronous
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.MakeStatementAsynchronous), Shared]
+    internal class CSharpMakeStatementAsynchronousCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        // error CS8414: foreach statement cannot operate on variables of type 'IAsyncEnumerable<int>' because 'IAsyncEnumerable<int>' does not contain a public instance definition for 'GetEnumerator'. Did you mean 'await foreach'?
+        // error CS8418: 'IAsyncDisposable': type used in a using statement must be implicitly convertible to 'System.IDisposable'. Did you mean 'await using' rather than 'using'?
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create("CS8414", "CS8418");
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            var constructToFix = TryGetStatementToFix(node);
+            if (constructToFix == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = editor.OriginalRoot;
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+                var statementToFix = TryGetStatementToFix(node);
+                if (statementToFix != null)
+                {
+                    MakeStatementAsynchronous(document, editor, statementToFix);
+                }
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private static void MakeStatementAsynchronous(Document document, SyntaxEditor editor, SyntaxNode statementToFix)
+        {
+            SyntaxNode newStatement;
+            switch (statementToFix)
+            {
+                case ForEachStatementSyntax forEach:
+                    newStatement = forEach
+                        .WithForEachKeyword(SyntaxFactory.Token(SyntaxKind.ForEachKeyword))
+                        .WithAwaitKeyword(SyntaxFactory.Token(SyntaxKind.AwaitKeyword).WithLeadingTrivia(forEach.GetLeadingTrivia()));
```

Comment:
nit: reuse the foreach-keyword.  just with the leading trivia stripped.  i.e. `.forEach.ForEachKeyword.WithoutLeadingTrivia()`.  This will be slightly more correct if there is trailing trivia on the keyword.

Reply:
There's not `WithoutLeadingTrivia()` on `SyntaxToken`. I'll look around for an alternative.

---
In reply to: [261064846](https://github.com/dotnet/roslyn/pull/33463#discussion_r261064846) [](ancestors = 261064846)

Reply:
Well, you can just do `.WithLeadingTrivia()`.  :)